router/default.go

```diff
@@ -714,6 +730,12 @@ func (r *router) flushRouteEvents(evType EventType) ([]*Event, error) {
 		return nil, fmt.Errorf("failed listing routes: %s", err)
 	}
 
+	// Do not advertise anything
+	if r.options.Advertise == AdvertiseNone {
```

Comment:
Shouldnt this be called before the `for` loop which iterates over the stored adverts?

Reply:
Good spot